### PR TITLE
refactor(builtins): centralize setting up builtins

### PR DESCRIPTION
**Pull request by LostNeophyte**
this pr adds a `setup_builtin` function that calls `builtin.setup()` and `setup` to have it in one place.
 This will make the next refactor(next version of https://github.com/LunarVim/LunarVim/pull/3776) easier

### CODE REVIEW

**Comment by opalmay** [comment]
I have an error loading a telescope extension 
image

**Comment by LostNeophyte** [comment]
Oh, I know what's the problem, will fix it

File: lua/lvim/core/builtins/init.lua [hunk 1]
```diff
@@ -23,12 +23,28 @@ local builtins = {
   "lvim.core.mason",
 }
 
+---@class LvimBuiltin
+---@field active boolean is builtin enabled
+---@field on_config_done function function called after the builtin is set up
+
 function M.config(config)
```
Comment (kylo252) [inline lua/lvim/core/builtins/init.lua:30]:
```suggestion
function M.init(config)
```

File: lua/lvim/core/builtins/init.lua [hunk 2]
```diff
@@ -23,12 +23,28 @@ local builtins = {
...
+  lvim.builtin = {}
+
   for _, builtin_path in ipairs(builtins) do
     local builtin = reload(builtin_path)
 
     builtin.config(config)
   end
 end
 
+function M.setup_builtin(modname)
```
Comment (kylo252) [inline lua/lvim/core/builtins/init.lua:41]: no need to repeat redundant name

```suggestion
function M.setup(modname)
```

File: lua/lvim/core/builtins/init.lua [hunk 3]
```diff
@@ -23,12 +23,28 @@ local builtins = {
...
+  local plugin = require("lvim.core." .. modname).setup()
+
+  local builtin_tbl = lvim.builtin[string.gsub(modname, "-", "_")]
+  if type(builtin_tbl.on_config_done) == "function" then
```
Comment (kylo252) [inline lua/lvim/core/builtins/init.lua:45]: you're not checking if `setup` has returned correctly

```suggestion
  if plugin and type(builtin_tbl.on_config_done) == "function" then
```

File: lua/lvim/core/builtins/init.lua [hunk 4]
```diff
@@ -23,12 +23,28 @@ local builtins = {
...
     local builtin = reload(builtin_path)
 
     builtin.config(config)
   end
 end
 
+function M.setup_builtin(modname)
+  local plugin = require("lvim.core." .. modname).setup()
+
+  local builtin_tbl = lvim.builtin[string.gsub(modname, "-", "_")]
```
Comment (kylo252) [inline lua/lvim/core/builtins/init.lua:44]:
```suggestion
  local builtin_tbl = lvim.builtin[modname:gsub("-", "_")]
```

**Review by kylo252 — COMMENTED** [review]
there are some ideas that we may want to resurrect from https://github.com/LunarVim/LunarVim/pull/1406

specifically: https://github.com/Tastyep/LunarVim/commit/d381444b70b9b27a211fe1724515e93aa07b4776

File: lua/lvim/core/builtins/init.lua (hunk 3 above)
Comment (LostNeophyte) [reply lua/lvim/core/builtins/init.lua:45]: not all builtins return a plugin (commands for example)

Comment (kylo252) [reply lua/lvim/core/builtins/init.lua:45]: that's why I mentioned that older PR to cleanly separate builtins from core, you could also check the table at the top of this file.

**Pull request closed without merging**